Meta RFC: Clarify expectations about the RFC process (#200)
reactjs/rfcs: README.md

  - Additions only likely to be _noticed by_ other implementors-of-React,
  invisible to users-of-React.

## What to expect

It is hard to write an RFC that would get accepted. Nevertheless, this shouldn't
discourage you from writing one.

React has a very limited API surface area, and each feature needs to work seamlessly with all other features.
Even among the team members who work on React full time every day, ramping up
and gaining enough context to write a good RFC takes more than a year.

In practice, React RFCs serve two purposes:

* **React Team RFCs** are submitted by [React Team members](https://reactjs.org/community/team.html) after extensive (sometimes,

> multi-month or multi-year) design, discussion, and experimentation. In practice, they comprise

Reviewer: ```suggestion
multi-month or multi-year) design, discussion, and experimentation. In practice, they comprise the
```

majority of the RFCs that got merged so far. The purpose of these RFCs is to preview the design
for the community and to provide an opportunity for feedback. We read every comment on the RFCs
we publish, respond to questions, and sometimes incorporate the feedback into the proposal.
Since our time is limited, we don't tend to write an RFC for a React feature unless we're very
confident that it fits the design. Although it might look like most React Team RFCs easily

> get accepted, in practice it's because 98% of ideas were left on the cutting floor. The remaining

Reviewer 2: This is a good thing to call out. 👍🏼

Reviewer 3: Nit, but I think the phrase is "cutting room floor".
```suggestion
get accepted, in practice it's because 98% of ideas were left on the cutting room floor. The remaining

```

2% that we feel very confident and have team consensus on about are the ones that we announce as RFCs for community feedback.

* **Community RFCs** can be submitted by anyone. In practice, most community RFCs do not get merged.
The most common reasons we reject an RFC is that it has significant design gaps or flaws, does not work
cohesively with all the other features, or does not fall into our view of the scope of React. However,
getting merged is not the only success criteria for an RFC. Even when the API design does not match
the direction we'd like to take, we find RFC discussions very valuable for research and inspiration.

> We don't always review community RFCs timely, but whenever we start work on a related area, we check

Reviewer 3: ```suggestion
We don't always review community RFCs in a timely manner, but whenever we start work on a related area, we check

```

the RFCs in that area, and review the use cases and concerns that the community members have posted.
When you send an RFC, your primary goal should not be necessarily to get it merged into React as is,
but to generate a rich discussion with the community members. If your proposal later becomes accepted,
that's great. But even if it doesn't, it won't be in vain. The resulting discussion often informs the next
proposal in the same problem space, whether it comes from the community or from the React Team. Many library
authors are reading the discussions, so RFCs often lead to community experimentation and userland solutions.

We apply the same level of rigour both to React Team RFCs and Community RFCs. The primary difference

> between them is in the design phase: React Team RFCs tend to be submitted at the end of the design

Reviewer 2: This too 👍🏼

## Reviewing RFCs

Currently, the React Team cannot commit to reviewing RFCs in a timely manner.
When you submit an RFC, your primary goal should be to solicit community feedback

> and generate a rich discussion. The React Team reevaluates the current list of

Reviewer: ```suggestion
and generate a rich discussion. The React Team re-evaluates the current list of

```

projects and priorities every several months. Even if an RFC is well-designed,
we often can't commit to integrating it right away. However, we find it very
valuable to revisit the open RFCs every few months, and see if anything catches
our eye. Whenever we start working on a new problem space, we also make sure
to check for prior work and discussion in any related RFCs, and engage with them.

We read all RFCs within a few weeks of submission. If we think the design fits React well,
and if we're ready to evaluate it, we will try to review it sooner. If we're hesitant about
the design or if we don't have enough information to evaluate it, we will leave it open
until it receives enough community feedback. We recognize it is frustrating to not receive
a timely review, but you can be sure that none of the work you put into an RFC is in vain.

## Inspiration

> React's RFC process owes its inspiration to the [Yarn RFC process], [Rust RFC process], and [Ember RFC process]

Reviewer 2: super nit: Missing trailing puncuation
